src/list.js

```diff
@@ -130,7 +130,7 @@ class TypeInferedList extends BaseImmutableList {
   }
 
   get(index, notSetValue) {
-    return this[$store] ? this[$store].get(index, notSetValue) :
+    return this[$store] ? this[$store].get(parseInt(index), notSetValue) :
```

Comment:
Oh my. This one's strange, we're either working around a bug or undefined behavior.

I think we should error if you try to `list.get('1a2b3c')`, but I'm also ok with merging as it is. @udnisap @davecoates any opinions?